### PR TITLE
Add authorization support based on allowed actions

### PR DESCRIPTION
**Pull request by jillcary**
# Type of change? 

- [ ] Bug fix
- [x] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 

- Get the user role info from JWT token
- Query authentication service to get allowed actions defined in role policies
- Enforce the policy of allowed actions

Related PR:
https://github.com/Azure/iothub-manager-dotnet/pull/131

**Checklist:**

- [x] All tests passed
- [x] The code follows the code style and conventions of this project
- [x] The change requires a change to the documentation
- [x] I have updated the documentation accordingly

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,6 +45,7 @@ Follow the instructions for
 
 This service depends on the following repository.
 1. [Storage Adapter Microservice](https://github.com/Azure/pcs-storage-adapter-dotnet)
+2. [Authentication Microservice](https://github.com/Azure/pcs-auth-dotnet) (if auth is enabled)
```
[the remark this answers is not in the document]
Comment (jillcary) [reply README.md]: Updated in the new repo:
https://github.com/Azure/remote-monitoring-services-dotnet/pull/14

File: WebService/v1/Filters/AuthorizeActionFilterAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        }
+
+        public async Task OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
+        {
+
+            bool isAuthorized = this.ValidateAuthorization(context.HttpContext, this.allowedAction);
+
+            if (!isAuthorized)
+            {
+                throw new NoAuthorizationException($"Current user is not authorized to perform this action: '{this.allowedAction}'");
```
Comment (zhang-hua) [inline WebService/v1/Filters/AuthorizeActionFilterAttribute.cs]: NotAuthorizedException is better here, as you suggested.

File: Services/External/UserApiModel.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.IoTSolutions.DeviceTelemetry.Services.External
+{
+    public class UserApiModel
```
Comment (zhang-hua) [inline Services/External/UserApiModel.cs]: This model is not used for now. should we remove it? same question for my code in Iothub manager.

**Comment by jillcary** [comment]
This PR has been migrated to the new repository: Azure/remote-monitoring-services-dotnet#14

**Pull request closed without merging**